src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -0,0 +1,514 @@
+        if (schema_object->getValue<int>("schema-id") != current_schema_id)
+            throw Exception(ErrorCodes::UNSUPPORTED_METHOD, "Cannot read Iceberg table: the table schema has been changed at least 1 time, reading tables with evolved schema is not supported");
+
+        avro::NodePtr root_node = manifest_file_reader->dataSchema().root();
+        size_t leaves_num = root_node->leaves();
+        size_t expected_min_num = format_version == 1 ? 3 : 2;
+        if (leaves_num < expected_min_num)
+        {
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS,
+                "Unexpected number of columns {}. Expected at least {}",
+                root_node->leaves(), expected_min_num);
+        }
+
+        avro::NodePtr data_file_node = root_node->leafAt(static_cast<int>(leaves_num) - 1);
+        if (data_file_node->type() != avro::Type::AVRO_RECORD)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "The parsed column from Avro file of `data_file` field should be Tuple type, got {}",
+                data_file_node->type());
+        }
+
+        data_type = AvroSchemaReader::avroNodeToDataType(data_file_node);
+        columns = parseAvro(*manifest_file_reader, data_type, manifest_path, getFormatSettings(getContext()));
+        const auto & tuple_type = assert_cast<const DataTypeTuple &>(*data_type.get());
+        const auto * tuple_column = typeid_cast<ColumnTuple *>(columns.at(0).get());
+
+        ColumnPtr file_path_column = tuple_column->getColumnPtr(tuple_type.getPositionByName("file_path"));
+
+        if (file_path_column->getDataType() != TypeIndex::String)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "The parsed column from Avro file of `file_path` field should be String type, got {}",
+                file_path_column->getFamilyName());
+        }
+
+        const auto * file_path_string_column = assert_cast<const ColumnString *>(file_path_column.get());
+
+        ColumnPtr content_column;
+        const ColumnInt32 * content_int_column = nullptr;
+        if (format_version == 2)
+        {
+            content_column = tuple_column->getColumnPtr(tuple_type.getPositionByName("content"));
+            if (content_column->getDataType() != TypeIndex::Int32)
+            {
+                throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "The parsed column from Avro file of `content` field should be Int type, got {}",
+                    content_column->getFamilyName());
+            }
+
+            content_int_column = assert_cast<const ColumnInt32 *>(content_column.get());
+        }
+
+        for (size_t i = 0; i < tuple_column->size(); ++i)
+        {
+            if (format_version == 2)
+            {
+                Int32 content_type = content_int_column->getElement(i);
+                if (content_type != 0)
```

Comment:
a comment could be useful
```suggestion
                /// 0=data, 1=position deletes, 2=equality deletes; default to 0 when reading v1 manifests
                if (content_type != 0)
```

Reply:
I'd better add a enum for all content types with the same values